storage/tables/data-types/index.md

```diff
@@ -993,29 +992,127 @@ The `TIMESTAMP` base type represents a date value with a time portion.
+    "name" VARCHAR(255),
+    "created_at" TIMESTAMP_NTZ
+);
+
+-- insert some data
+INSERT INTO "typed_table"
+VALUES
+    (1, '75', '2020-01-01 00:00:00');
+
+-- create another table with datatypes based on an existing table
+CREATE OR REPLACE TABLE "typed_table_2" AS
+SELECT
+    "id"::varchar AS "string_id",
+    "name"::number AS "numeric_name",
+    "created_at"::date AS "typed_date"
+FROM
+    "typed_table";
+```
+
+***Note:** The data type hinting is the components' responsibility, so components must be updated by their respective authors to support this. The database extractors that are maintained by Keboola already provide datatypes.  There is no list of components that support this feature. You may check the component's documentation to see if it supports native data types.* 
+
+### How to Define Data Types
+
+#### Using actual data types of the storage backend
+
+For example, in the case of Snowflake, you can create a column of type `TIMESTAMP_NTZ` or `DECIMAL(20,2)`. This allows you to specify all the data type details, including precision and scale, for instance. But it's tied to the specific storage backend, and thus it's not portable.
+
+An example of such a column definition in a table-definition API endpoint call is as follows:
+
+```json
+{
+  "name": "id",
+  "definition": {
+    "type": "DECIMAL",
+      "length": "20,2",
+      "nullable": false,
+      "default": "999"
+  }
+}
+```
+
+#### Using Keboola-provided [BASE TYPES](#base-types)
+
+Specifying native types using [BASETYPES](#base-types) is ideal for component provided types as they are storage backend agnostic. However, they can be used for the table-definition API endpoint as well. The definition is as follows:
+
+```json
+{
+  "name": "id",
+  "basetype": "NUMERIC"
+}
+```
+
+### Changing Types of Existing Typed Columns
+
+**You can't change the type of column of a typed table once it has been created.** There are multiple ways to work around this. 
+
+First, if the table is loaded using full load, you can drop the table and create a new table with the correct types and load the data there. 
+
+If the table is loaded incrementally, you must create a new column and copy the data from the old one.
+
+* You have a column `date` of type `VARCHAR` in a typed table, and you want to change it to `TIMESTAMP`.
+* You first add a new column `date_timestamp` of type `TIMESTAMP` to the table.
+* Then you change all the jobs filling the table to fill the new and old columns.
+* Then you run an ad-hoc transformation, which will copy data from `date` to `date_timestamp` in the existing rows.
+* Then you can slowly change all the places where `date` is used to use `date_timestamp` instead.
+* When you only use the new column, the old one can be removed.
+
+In both cases, check all the downstream configurations so that you don't get any schema mismatch. This is especially important for data destinations, where there is already an existing table in the destination.

```

Comment:
```suggestion
In both cases, check all the other configurations using the table so that you don't get any schema mismatch. This is especially important for data destination components (writers), where there is already an existing table in the destination.
```